x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -917,6 +1007,20 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("assigned", assigned);
             builder.field("assigned_shards", assignedShardIds.stream().limit(MAX_AMOUNT_OF_SHARDS).toList());
             builder.field("assigned_shards_count", assignedShardIds.size());
+            if (unassignedAllocationResults != null && unassignedAllocationResults.size() > 0) {
```

Comment:
This can be simplified with `builder.xContentList("unassigned_allocation_results", unassignedAllocationResults);` and I think we should serialize it as long as it is not null.